File: aparcamiento/IvanRC-aparcamiento/IvanRC-aparcamiento-Java/src/main/java/ModelsAparcamiento/Vehiculo.java
package ModelsAparcamiento;

import java.util.Objects;
import java.util.Scanner;

public class Vehiculo {

    private static Scanner sc = new Scanner(System.in);

    public Conductor conductor;
    public int id;
    private String matricula;
    private int añoFabricacion;
    private tipoVehiculo tipo;

    private static int contador = 1;
    private int nextId(){
        return contador++;
    }

    public Vehiculo(String matricula, int añoFabricacion, tipoVehiculo tipo, Conductor conductor) {
        this.id = nextId();
        this.matricula = matricula;
        this.añoFabricacion = añoFabricacion;
        this.tipo = tipo;
        this.conductor = conductor;
    }

    public String getMatricula() {
        return matricula;
    }

    public int getAñoFabricacion() {
        return añoFabricacion;
    }

    public String getTipo() {
        String tipoV = null;
        switch (this.tipo){
            case coche -> tipoV = "coche";
            case moto -> tipoV = "moto";
            case patinete -> tipoV = "patinete";
        }
        return tipoV;
    }

    public Conductor getConductor() {
        return conductor;
    }

    @Override
    public String toString() {
        return "Vehiculo{" +
                "conductor=" + conductor +
                ", matricula='" + matricula + '\'' +
                ", añoFabricacion=" + añoFabricacion +
                ", tipo='" + tipo + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Vehiculo vehiculo = (Vehiculo) o;
        return añoFabricacion == vehiculo.añoFabricacion && Objects.equals(conductor, vehiculo.conductor) && Objects.equals(matricula, vehiculo.matricula) && Objects.equals(tipo, vehiculo.tipo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(conductor, matricula, añoFabricacion, tipo);
    }
}

File: aparcamiento/IvanRC-aparcamiento/IvanRC-aparcamiento-Kotlin/src/main/kotlin/ModelsAparcamiento/Vehiculo.kt
package ModelsAparcamiento

data class Vehiculo(val matricula: String, val añoFabricacion: Int, val tipo: tipoVehiculo, val conductor: Conductor) {
    val id: Int = nextId()

    companion object{
        private var contador = 1
        fun nextId(): Int {
            return contador++
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Vehiculo

        if (matricula != other.matricula) return false
        if (añoFabricacion != other.añoFabricacion) return false
        if (tipo != other.tipo) return false
        if (conductor != other.conductor) return false

        return true
    }

    override fun hashCode(): Int {
        var result = matricula.hashCode()
        result = 31 * result + añoFabricacion
        result = 31 * result + tipo.hashCode()
        result = 31 * result + conductor.hashCode()
        return result
    }

    override fun toString(): String {
        return "Vehiculo(matricula='$matricula', añoFabricacion=$añoFabricacion, tipo=$tipo, conductor=$conductor, id=$id)"
    }
}